packages/samples/headless-ssr-commerce/app/_components/cart.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Cart as CartController,
+  CartItem,
+  CartState,
+  ContextState,
+} from '@coveo/headless/ssr-commerce';
+import {useEffect, useState, FunctionComponent} from 'react';
+import {formatCurrency} from '../_utils/format-currency';
+
+interface CartProps {
+  staticState: CartState;
+  controller?: CartController;
+  staticContextState: ContextState;
+}
+
+export const Cart: FunctionComponent<CartProps> = ({
+  staticState,
+  controller,
+  staticContextState,
+}) => {
+  const [state, setState] = useState(staticState);
+
+  useEffect(
+    () => controller?.subscribe(() => setState({...controller.state})),
+    [controller]
+  );
+
+  const adjustQuantity = (item: CartItem, delta: number) => {
+    controller?.updateItemQuantity({
+      ...item,
+      quantity: item.quantity + delta,
+    });
+  };
+
+  const isCartEmpty = () => {
+    return state.items.length === 0;
+  };
+
+  const purchase = () => {
+    controller?.purchase({id: crypto.randomUUID(), revenue: state.totalPrice});
+  };
+
+  const emptyCart = () => {
+    controller?.empty();
+  };
+
+  const language = () => staticContextState.language;
+  const currency = () => staticContextState.currency;
```

Comment:
Should we also pass the context controller as a prop and subscribe to its state so that the language and currency will be updated in the cart component if it changes after initialization?

Reply:
yes, I will update that